### PR TITLE
Bot versioning 

### PR DESCRIPTION
**Pull request by OtherwiseJunk**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -174,11 +190,16 @@
...
           }
         }
       }
     }
   },
-  "defaultProject": "hmct-web"
-}
+  "defaultProject": "hmct-web",
+  "cli": {
+    "analytics": "28232fc6-98fc-42ca-9299-71ceb6122e0a"
```
Comment (jonnhy12333) [inline angular.json]: Looks like this is the only real change in this file, is this change necessary? not sure what it does...

File: src/bots/services/bot-base/bot-base.service.ts [hunk 2]
```diff
@@ -24,13 +25,14 @@ export abstract class BotBaseService {
...
 
-  constructor(key: string, displayName: string, imageName: string, protected botService: BotService, isHidden?: boolean) {
+  constructor(key: string, displayName: string, imageName: string, protected botService: BotService, isHidden?: boolean, version: string = "") {
     this.$key = key;
     this.displayName = displayName;
     this.imageUrl = `bots/assets/images/${imageName}`;
     this.mood$ = botService.getMood$(this);
     this.isActive = !!environment.bots.find(b => b === key);
     this.isHidden = isHidden;
+    if (version) console.log(`Starting up ${displayName} Version ${version}`);
```
Comment (jonnhy12333) [inline src/bots/services/bot-base/bot-base.service.ts]: I think for this new parameter, it should probably be to the _left_ of 'isHidden' since it's more important.  I think isHidden is the most likely parameter to be left out / should be right-most:

constructor(key: string, displayName: string, imageName: string, protected botService: BotService, version?: string, isHidden?: boolean)

Also you can use the '?' syntax for the optional parameter instead of the = "" to save some characters.

File: angular.json (hunk 1 above)
Comment (OtherwiseJunk) [reply angular.json]: Vsc kept doing this automatically, i must have committed it by mistake.

File: src/bots/services/bot-base/bot-base.service.ts (hunk 2 above)
Comment (OtherwiseJunk) [reply src/bots/services/bot-base/bot-base.service.ts]: Ye seems good to me.

Comment (jonnhy12333) [reply src/bots/services/bot-base/bot-base.service.ts]: oh I missed uh.... we should probably set `this.version = version` to actually hang on to this value or else there's no reason for the 'version' prop on the class

Comment (OtherwiseJunk) [reply src/bots/services/bot-base/bot-base.service.ts]: Agreed, I'm actually just gonna remove the property. We don't really care we just want to call it out on creation, I don't think we'd ever use it.
